rs/replicated_state/src/metadata_state.rs

```diff
@@ -791,7 +791,7 @@ impl SystemMetadata {
             // we used CURRENT_CERTIFICATION_VERSION here, the state hash would
             // NOT be guaranteed to be constant, potentially leading to
             // hard-to-track bugs in state manager.
```

Comment:
I just changed this here to see what happens. I had to change two hardcoded cashes for a test in state manager. Otherwise everything appears to be working, at least according to the CI.

Reply:
FWIW, as long as you sort out the details with the Consensus team and ensure that this doesn't break subnet splitting, the change LGTM.

Reply:
I asked in the consensus channel and they didn't see an immediate problem.
I ran this [system test](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/tests/consensus/subnet_splitting_test.rs) manually. It passed. AFAICT this should inspire enough confidence for an approval.